packages/trackerless-network/src/logic/neighbor-discovery/Handshaker.ts

```diff
@@ -78,11 +78,11 @@ export class Handshaker {
 
     private selectParallelTargets(excludedIds: DhtAddress[]): HandshakeRpcRemote[] {
         const neighbors = this.config.nearbyNodeView.getFirstAndLast(excludedIds)
-        while (neighbors.length < PARALLEL_HANDSHAKE_COUNT && this.config.randomNodeView.size(excludedIds) > 0) {
-            const random = this.config.randomNodeView.getRandom(excludedIds)
-            if (random) {
-                neighbors.push(random)
-            }
+        const excludeFromRandomView = [...excludedIds, ...neighbors.map((neighbor) => getNodeIdFromPeerDescriptor(neighbor.getPeerDescriptor()))]
```

Comment:
Could simplify the state: this can always be calculated from `excludedIds` and `neighbors`.

Reply:
Not sure I follow @teogeb ?

Reply:
@juslesan The point is that we don't need the `excludeFromRandomView` variable as it can be calculated on-demand:

`const getExcludeFromRandomView = () => [...excludedIds, ...neighbors.map((neighbor) => getNodeIdFromPeerDescriptor(neighbor.getPeerDescriptor()))]`

It can be a local helper function inside the `selectParallelTargets` method.